server/controllers/mentionController.js

```diff
@@ -68,6 +69,7 @@ const getMentions = async (req, res) => {
           }
         });
       }
+      // after setting filteredMentions, nothing is done with it
 
```

Comment:
I think after setting filtereMentions, paginatedMentions should be a slice of it instead of allMentions

Reply:
Yep, I agree. Let's add this code